edge-util/src/Microsoft.Azure.Devices.Edge.Util/PersistentStorageValidation.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Util
+{
+    using System.IO;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public static class PersistentStorageValidation
+    {
+        public static void ValidateStorageIdentity(string storagePath, string deviceId, string iotHubHostname, string moduleId, Option<string> moduleGenerationId)
+        {
+            ILogger logger = SetupLogger();
+            DeviceIdentity currentIdentity = new DeviceIdentity(deviceId, iotHubHostname, moduleId, moduleGenerationId);
+            string filepath = Path.Combine(storagePath, "DEVICE_IDENTITY");
+            string json = JsonConvert.SerializeObject(currentIdentity);
+
+            if (!File.Exists(filepath))
+            {
+                File.WriteAllText(filepath, json);
+            }
+            else
+            {
+                DeviceIdentity savedIdentity = JsonConvert.DeserializeObject<DeviceIdentity>(File.ReadAllText(filepath));
+
+                if (currentIdentity.Equals(savedIdentity))
+                {
+                    logger.LogInformation("Persistent storage is for a different device identity {actual}, then the current {current}. Deleting local storage.",  savedIdentity.DeviceId, currentIdentity.DeviceId);
+                    Directory.Delete(storagePath, true);

```

Comment:
why not simply delete the file . Seems an expensive operation to first delete the directory then create a directory

Reply:
Everything within this directory is related to the previous device identity. That's why we delete the directory and everything within it.